zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/identity/AuthorizationCheckBehavior.java

```diff
@@ -464,13 +461,16 @@ public AuthorizedTenants getAuthorizedTenantIds(final TypedRecord<?> command) {
         getPersistedMappings(command).stream()
             .flatMap(
                 mapping -> {
-                  final var tenantIdsList = mapping.getTenantIdsList();
+                  final var tenantIds =
+                      new ArrayList<>(
+                          membershipState.getMemberships(
+                              EntityType.MAPPING, mapping.getMappingId(), RelationType.TENANT));
                   final var groupIds =
                       mapping.getGroupKeysList().stream()
                           .map(key -> Long.toString(key))
                           .collect(Collectors.toSet());
-                  tenantIdsList.addAll(getTenantIdsForGroups(groupIds));
-                  return tenantIdsList.stream();
+                  tenantIds.addAll(getTenantIdsForGroups(groupIds));
+                  return tenantIds.stream();
```

Comment:
🔧 As we're dealing with lists and we ultimately want a stream at the end of it I believe this can simplified to something like 

``` 
mapping -> Stream.concat(
  membershipState.getMemberships(
      EntityType.MAPPING, mapping.getMappingId(), RelationType.TENANT).stream(),
  mapping.getGroupKeysList().stream().map(key -> Long.toString(key)))
```

Reply:
Done in abc5d31